File: b0316curr/app/src/main/java/com/example/b0316curr/MainActivity.java
package com.example.b0316curr;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.provider.DocumentsContract;
import android.util.Log;
import android.widget.TextView;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {
    private Document document;
    private Thread thread;
    private Runnable runnable;
    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = (TextView)findViewById(R.id.textView);

        textView.setText("OK2");
        Log.d("MyLog", "765");
//        Jsoup
        init();


    }

    private void init(){
        runnable = new Runnable() {
            @Override
            public void run() {
                getWeb();
            }
        };
        thread = new Thread(runnable);
        thread.start();
    }

    private void getWeb(){
        try {
            Log.d("MyLog", "1234");
            document = Jsoup.connect("https://minfin.com.ua/currency/").get();
            Log.d("MyLog", "Title : " + document.title());
            String string1 = "Title : " + document.title();
            runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    textView.setText(string1);
                }
            });

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: l2162pref/app/src/main/java/com/example/freon/l2162pref/MainActivity.kt
package com.example.freon.l2162pref

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    var pref: SharedPreferences? = null

    var editText: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editText = findViewById(R.id.editText)
        pref = getSharedPreferences("Text", Context.MODE_PRIVATE)

        var s = pref?.getString("Text", "")
        editText?.setText(s)

    }

    override fun onDestroy() {
        super.onDestroy()
        var s : String = editText?.text.toString()
        val editor = pref?.edit()
        editor?.putString("Text", s)
        editor?.apply()

    }
}